### ISSUE TITLE
Android - TextInput Autofill not working 

### ISSUE THREAD

**Issue by todoranalex** [software-mansion/react-native-screens#349]
"react-native": "0.61.4"
"react-native-screens: "2.0.0-alpha.5"

It seems that the library blocks the native autofill suggestions view. Also, there is no save password dialog after a successful login. (This dialog is auto popped-up by the Android framework). 

```
<TextInput
              placeholder={'Username'}
              autoCompleteType={'email'}
 />
  <TextInput
              placeholder={'Password'}
              autoCompleteType={'password'}
              secureTextEntry={true}
/> 
```
Setting those props should be enough to trigger the autofill mechanism. 

What we tried and **did not work**:
- Update to a library version after the PR from [#205](https://github.com/software-mansion/react-native-screens/issues/205) was merged.
- Set the TextInput width to 99% on first render and then to 100%, as suggested 
- Set the TextInput editable prop to false on first render and then on true, as suggested

**Comment by WoLewicki** [comment]
@todoranalex can you repro the issue in the newest version?

**Comment by ovy9086** [comment]
yes. this is still happening with RN 0.62 , and screens 2.4.0

**Comment by WoLewicki** [comment]
Can you provide a repo/snack with the minimal configuration needed to repro the issue?

**Comment by WoLewicki** [comment]
I am closing it due to no response in more than 30 days. Feel free to comment to reopen it.

**Comment by mstrk** [comment]
Hi @WoLewicki I can confirm that by using `enableScreens`, the issues described by the OP happen in a real device. If one do not call `enableScreens` the native secondary popups (which is the case of pick saved password) do appear. I did not tested in the emulator tho.

**Comment by WoLewicki** [comment]
@mstrk can you provide a repo/snack with minimal configuration needed to reproduce the issue?

**Comment by mstrk** [comment]
@WoLewicki I was trying to reproduce it in a snack and I can’t 🤔

In the project with the issue I don’t use expo btw. I'll try find some time to create a repo to test it without expo.

**Comment by ovy9086** [comment]
@WoLewicki  here's a repo project for this : https://github.com/ovy9086/rn-autofill-bug

It's a bit weird but now it seems that the 'Save password' pop-up doesn't show at all when using react-nav-5 , no matter if screens is enabled or not.

But still, the context menu to autofill the inputs, is not working when screens are enabled.

As you can see in the screenshots, when screens are enabled, I get a toast 'Contents can't be auto-filled' . When screens are disabled, the autofill context menu is shown.

https://imgur.com/a/nuxYb8y

**Comment by mstrk** [comment]
Hi @ovy9086 thanks for the repo. I've tested it in the same android device I have the problem and like the snack I tried to reproduce the issue, this repo also works fine :s

I get the autocomplete and save password popups. So I’m really confused now.

**Comment by todoranalex** [comment]
I tested @ovy9086 's repo on a Pixel 2 and the autofill framework works fine for me with screens turned OFF. If screens is turned ON, then I can't autofill the input field and also no pop-up to save your password is shown. But I also saw the following:
 
turn screens ON. 
Open App -> Try to autofill email -> you get 'Content's can't be autofilled'. Put App in OVERVIEW (or background) -> Resume -> Try to autofill again email -> Now you see the autofill suggestions on the email field. 

Then, I tried to put the app in overview exactly after pressing the Login button, and then resume it while it transitions to the next screen, and I could see for like 0.5 seconds the Password Save Pop-up, then it disappeared.  

So, most likely, the screens library somehow cuts off those elements..

**Comment by ovy9086** [comment]
@mstrk @todoranalex yeah sorry for the confusions. I have left screens off in my repo. updated it now so it's enabled by default 🙈

**Comment by WoLewicki** [comment]
Ok, I am reopening it then.

**Comment by mstrk** [comment]
@ovy9086 @todoranalex absolutely right! I didn’t noticed the `false` argument, I normally comment the function, wasn’t aware that we can pass arguments sorry.

Tested again and yes now I can’t see any of the popups.

**Comment by bjjeong** [comment]
Still occurring in `react-native-screens@2.11.0` and `react-native@63.2`

**Comment by kmagiera** [comment]
Thanks @ovy9086 for providing a repro, however I can't seem to be able to reproduce the problem there. When running your repro app on Pixel 2 with react-native@0.63.2 I'm getting "Contents can't be autofilled" message regardless of whether screens are enabled or not.

Next, I tried to remove react-navigation out of the equation and put the text fields from your app to one of out examples here: https://github.com/software-mansion/react-native-screens/blob/master/Example/nativeStack.js – after doing that I could use autofill without any problems.

You commented later that the autofil does not work regardless of screens on/off option. I assume this means that the problem could be on the React-Navigation side and not because of react-native-screens. I'll let @WoLewicki double check that but it appears that the provided repro does not seem to showcase the issue for us.

**Comment by todoranalex** [comment]
@kmagiera I just tested @ovy9086 's repo on a PIXEL 2. With screens turned on, I get 'Contents can't be autofilled', and with screens turned OFF I'm able to autofill and also I get the Save Password pop-up after 'logging' in.

**Comment by ovy9086** [comment]
@kmagiera sorry for the confusions, re-tested this repo on Pixel 3A and :

Autofill Text Input : 
* `enableScreens(true)` 👉  Toast 'Contents can't be autofilled' 🔴 
* `enableScreens(false)` 👉  I am able to autofill the user and password as expected ✅ 

Autofill Save suggestion:
* `enableScreens(true)` 👉   Save password pop-up after login not shown 🤷  🔴 
* `enableScreens(false)` 👉  Save password pop-up after login not shown 🤷  🔴

**Comment by kmagiera** [comment]
Thanks for checking again, and sorry for the confusion. Previously I tried to pull your App.js into the screens Example app where I saw the behavior I descried earlier.

I can confirm that I see the same outcome as you when running the app from your repo. So this likely means we have some discrepancy between the dependencies or the project config that is causing different behavior.

On a flip side, I also tried using ScreenContainer and ScreenStack directly on your project from github by replacing your App.js with the nativeStack.js example and the autofil works there. This means that this is likely some issue related to the way react-navigation use screens rather than something specific to screens library.

**Comment by RupamShaw** [comment]
Its still autofill not working

**Comment by aperomingo** [comment]
I have suffered a lot to implement this in iOS successfully:
- Create account (password suggestion and later saved in the KeyChain)
- Login (choice of username and password of your KeyChain).

However, in Android there is little information and I think that it is not very successful:
- The information about the **React Native** <TextInput> only tells you to implement the property `autoCompleteType = {'password'}` and `importantForAutofill = {'yes' }`.
- Then on the other hand, on **Android** https://developer.android.com/guide/topics/text/autofill-services they tell you about "Permissions and manifest declarations" .
- Also, this collides with `react-native-screens` and its **enableScreens** method, which has not worked for me with either true or false.

I don't know what to do, these kinds of problems that always happen with React Native make me desperate.
Normally you have more problems and it is more complicated with iOS, but in this case I think not, in my case.

**Comment by WoLewicki** [comment]
Are the `TextInput`s nested in a `ScrollView` component (does not need to be direct nesting)? I tested it a bit and it looks like having a `ScrollView` as a parent makes the autofill not work correctly. Can someone elaborate on this?

**Comment by todoranalex** [comment]
For me it's not working even if the text inputs are not inside a scroll view. (Pixel 2)

**Comment by WoLewicki** [comment]
Could you check such an example? 

```js
import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  TextInput,
  StatusBar,
  View,
  Text,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {enableScreens} from 'react-native-screens';

enableScreens(false);

const LoginScreen = (props) => {
  return (
    <>
      <TextInput
        style={{fontSize: 20}}
        autoCompleteType={'username'}
        importantForAutofill={'yes'}
        placeholder={'Username'}
      />
      <TextInput
        style={{fontSize: 20}}
        autoCompleteType={'password'}
        importantForAutofill={'yes'}
        placeholder={'Password'}
        secureTextEntry={true}
      />
      <TouchableOpacity
        onPress={() => props.navigation.replace('Home')}
        style={{
          height: 60,
          backgroundColor: 'red',
          width: '100%',
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Text>LOGIN</Text>
      </TouchableOpacity>
      <ScrollView>
        <TextInput
          style={{fontSize: 20}}
          autoCompleteType={'username'}
          importantForAutofill={'yes'}
          placeholder={'Username'}
        />
        <TextInput
          style={{fontSize: 20}}
          autoCompleteType={'password'}
          importantForAutofill={'yes'}
          placeholder={'Password'}
          secureTextEntry={true}
        />
      </ScrollView>
    </>
  );
};

const HomeScreen = (props) => {
  return (
    <View
      style={{
        flex: 1,
        backgroundColor: 'red',
        alignItems: 'center',
        justifyContent: 'center',
      }}>
      <Text>
        Hello. Welcome. You should now be prompted to save your credentials.
      </Text>
      <TouchableOpacity
        onPress={() => props.navigation.replace('Login')}
        style={{
          height: 60,
          backgroundColor: 'red',
          width: '100%',
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Text>LOGIN</Text>
      </TouchableOpacity>
    </View>
  );
};

const Stack = createStackNavigator();

function MyStack() {
  return (
    <Stack.Navigator>
      <Stack.Screen name="Login" component={LoginScreen} />
      <Stack.Screen name="Home" component={HomeScreen} />
    </Stack.Navigator>
  );
}

export default function App() {
  return (
    <NavigationContainer>
      <MyStack />
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  body: {
    backgroundColor: Colors.white,
  },
});
```
In my case, when I have the `enableScreens(true)`, the `TextInput`s above the `Login` button have Autofill available, and the ones under the `Login` button do not (these have a `ScrollView` above them). If I have the `enableScreens(false)`, Autofill works in both options. Can someone check if it is only specific to my setup or is it like this for you too?

**Comment by limaAniceto** [comment]
@WoLewicki Appreciate you giving that example and the screens lib

We're facing this issue as well with our bump on `react-native-screens` (v2.18.1) and `react-navigation` (v5)

We used your [example above ](https://github.com/software-mansion/react-native-screens/issues/349#issuecomment-721742574) and both forms (the one inside the `scrollView` and the one outside of it) weren't autofillable. 

The image below is how we integrated it, and you can see the Toast `Contents can't be auto-filled`

![image](https://user-images.githubusercontent.com/2905749/110823654-3c146780-828a-11eb-861d-93873fefbf6b.png)

If you'd like us to try other cases, we're happy to do so

**Comment by WoLewicki** [comment]
I couldn't find anything that can cause the issue and I am a bit helpless on how to resolve it. I will gladly accept the examples of the cases where this works and where it doesn't to have better knowledge of the problem. It seems weird that it doesn't work for you in both cases. Are you using some other components above the screen shown? My example consisted of only the code shown above.

**Comment by foloinfo** [comment]
I had the same issue and accidentaly realized that it works as expected when I use state to mount the TextInput right after mouting the screen component.

example:

```js

const Screen = ()=> {

  const [shown, setShown] = useState(false)
  useEffect(()=> {
    setShown(true)
  }, [])

  return (
    <View>
      {shown ? (
        <>
          <TextInput
             autoCompleteType='email'
             importantForAutofill='yes'
             //...rest
           />
           <TextInput
             autoCompleteType='password'
             importantForAutofill='yes'
             secureTextEntry={true}
             //...rest
           />
         <>
       ) : null}
    </View>
  )

}

```

I'm using `"react-native-screens": "~3.0.0"` and react-navigation v5 with expo sdk 41 (React Native 0.63).
Also I have called `enableScreen(true)` and the TextInput is inside the ScrollView.

I have no idea why it solves the issue but at least it's working for me right now.
I hope this helps on the other versions.

**Comment by limaAniceto** [comment]
@foloinfo can confirm that this workaround also works for us

@WoLewicki this seems like a race condition and the extra re-render gives the time that the OS auto-fill feature needs to read the elements. 

Keen to hear your thoughts

**Comment by WoLewicki** [comment]
@limaAniceto yeah it might be the proper explanation. So it seems like there is nothing we can do on the side of `react-native-screens` other than suggest this workaround unfortunately.

**Comment by RupamShaw** [comment]
> I had the same issue on Android devices and accidentaly realized that it works as expected when I use state to mount the TextInput right after mouting the screen component.
> 
> example:
> 
> ```js
> const Screen = ()=> {
> 
>   const [shown, setShown] = useState(false)
>   useEffect(()=> {
>     setShown(true)
>   }, [])
> 
>   return (
>     <View>
>       {shown ? (
>         <>
>           <TextInput
>              autoCompleteType='email'
>              importantForAutofill='yes'
>              //...rest
>            />
>            <TextInput
>              autoCompleteType='password'
>              importantForAutofill='yes'
>              secureTextEntry={true}
>              //...rest
>            />
>          <>
>        ) : null}
>     </View>
>   )
> 
> }
> ```
> 
> I'm using `"react-native-screens": "~3.0.0"` and react-navigation v5 with expo sdk 41 (React Native 0.63).
> Also I have called `enableScreen(true)` and the TextInput is inside the ScrollView.
> 
> I have no idea why it solves the issue but at least it's working for me right now.
> I hope this helps on the other versions.

This works for me but new issue came just after logout or session expires returning back to login screen not showing autofill.. It only shows once going of to another app and autofill contents shows.

**Comment by schiller-manuel** [comment]
Can confirm that the workaround by https://github.com/software-mansion/react-native-screens/issues/349#issuecomment-856395868 works the first time, but when the screen is shown again (e.g. after logout), autofil does not work anymore.

**Comment by sregg** [comment]
Has anyone found a workaround that works all the time?
It'd be sad to disable those great native screens just for a small auto-fill issue...

**Comment by homosape** [comment]
This issue seems to be haunting us... I've tried all the mentioned workarounds but nothing worked for me. I am using React Native Paper's textinput and Formik.

Although, I've tested those workarounds with plain react native's textinput as well. 

I am on expo sdk 44, react navigation v6 ,react native screens 3.1+

**Comment by DineshKachhot** [comment]
> 

Same configuration and same issue I am facing with expo sdk 44.

**Comment by dlehddnjs** [comment]
I think this is still occurs. 
Is there any solution?

**Comment by kkafar** [comment]
I was not engaged in solving this problem initially, but tracking the discussion and understanding the issue it seems that, [as it was stated](https://github.com/software-mansion/react-native-screens/issues/349#issuecomment-860516260), not much can be done on `react-native-screens` side. I'll close the issue, however the discussion will remain open. If any new circumstances arise, please tag me.

**Comment by yaniv-blink** [comment]
@kkafar i don't think that the workaround is good enough, it doesn't work well in our case.
I also noticed that in some devices ( specifically devices who run Android 11 and below) it works very well. 
Maybe the fix worked in these devices, but it doesn't work on newer ones. 
Please reopen the case.

**Comment by kkafar** [comment]
I'll try to dive into it then

**Comment by dylanhillier** [comment]
I have been battling this also. Some observations:

1. If i got to the screen which has the email/password fields, i dont get prompted to auto-fill.
2. If I hold my finger on one of the fields, i get prompted to "Autofill", and when i hit the button, i get a message appear that the fields can't be auto filled.
3. If i use the work around proposed, i still don't get prompted.
4. I can occasionally get the auto-fill prompt in the following cases:
- If i click on the email field, and on the keyboard which comes up, i press the settings, which takes me to the keyboard settings, then i just hit the back button to go back to the original screen.
- If i leave the screen up with the email/password fields then i hit the button on the side of the phone to turn off the mobile screen, press that button again, login, then as the screen displays again i get prompted with autofill.

**Comment by syntax-e** [comment]
My team is experiencing the same thing. The google password manager prompt pops up when entering username and password. However the user cannot save credentials.

**Comment by gyl89** [comment]
Same issue, the workaround does not work at all for me

**Comment by gyl89** [comment]
> I have been playing with this quite a bit, and i have observed that if i change: From:
> 
> `import {TextInput} from 'react-native';` To:
> 
> `import {TextInput} from 'react-native-gesture-handler';`
> 
> The autofill starts working. I have not yet spent time investigating the why. But would be interested if this also resolves other peoples issues.

Doesn't work for me

**Comment by Eikjos** [comment]
Hi,
I had the same problem, I solved it by connecting a google account in the phone settings and allow autofill for credentials.
And it works

**Comment by nikolal** [comment]
We have the same issue.
Nothing from the proposed solutions works.

**Comment by Unn4m3DD** [comment]
If you are still having this issue try removing all other <TextInput/> components, for me the culprit was another TextInput being considered the target for autofill instead of the correctly tagged ones. If your issue is indeed with android targeting other TextInput's you can tag them as importantForAutofill="no" and the issue will go away.

Also some notes on testing this, I had much more luck testing with the autofill set to LastPass instead of Google Password Manager, it shows the save password prompt with usesClearTextTraffic="false" which is very useful when testing react native connected to metro server. You can then test GPM with a release build

Hope this is helpful!

**Comment by MarianaLima2000** [comment]
Update "react-native-screens" to "3.29.0" works for me

**Comment by MarianaLima2000** [comment]
Updating 'react-native-screens' to '3.29.0' worked for me

**Comment by SK-bool** [comment]
Try upgrading to 3.31.1, this seems to have resolved my issue. For reference: 

My <TextInput /> (username) 
 has autoComplete={'email'} 
  and <TextInput />(password) 
 has autoComplete={'current-password'}

**Comment by tboba** [comment]
Hi, all! Right now I'm testing repro provided below and I can't reproduce this issue anymore. It looks like autofilling works great right now! I'm testing an application with react-native-screens 3.31.0 and Google Pixel 6 (as a physical device).
Could you confirm that this issue is resolved in your applications?

<details>
<summary>Reproducer</summary>

```tsx
import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  TextInput,
  StatusBar,
  View,
  Text,
} from 'react-native';

import { Colors } from 'react-native/Libraries/NewAppScreen';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createStackNavigator } from '@react-navigation/stack';
import {
  GestureHandlerRootView,
  TouchableOpacity,
} from 'react-native-gesture-handler';
import { enableScreens } from 'react-native-screens';

enableScreens(false);

const LoginScreen = props => {
  return (
    <GestureHandlerRootView>
      <TextInput
        style={{ fontSize: 20 }}
        autoCompleteType={'username'}
        importantForAutofill={'yes'}
        placeholder={'Username'}
      />
      <TextInput
        style={{ fontSize: 20 }}
        autoCompleteType={'password'}
        importantForAutofill={'yes'}
        placeholder={'Password'}
        secureTextEntry={true}
      />
      <TouchableOpacity
        onPress={() => props.navigation.replace('Home')}
        style={{
          height: 60,
          backgroundColor: 'red',
          width: '100%',
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Text>LOGIN</Text>
      </TouchableOpacity>
      <ScrollView>
        <TextInput
          style={{ fontSize: 20 }}
          autoCompleteType={'username'}
          importantForAutofill={'yes'}
          placeholder={'Username'}
        />
        <TextInput
          style={{ fontSize: 20 }}
          autoCompleteType={'password'}
          importantForAutofill={'yes'}
          placeholder={'Password'}
          secureTextEntry={true}
        />
      </ScrollView>
    </GestureHandlerRootView>
  );
};

const HomeScreen = props => {
  return (
    <View
      style={{
        flex: 1,
        backgroundColor: 'red',
        alignItems: 'center',
        justifyContent: 'center',
      }}>
      <Text>
        Hello. Welcome. You should now be prompted to save your credentials.
      </Text>
      <TouchableOpacity
        onPress={() => props.navigation.replace('Login')}
        style={{
          height: 60,
          backgroundColor: 'red',
          width: '100%',
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Text>LOGIN</Text>
      </TouchableOpacity>
    </View>
  );
};

const Stack = createNativeStackNavigator();

function MyStack() {
  return (
    <Stack.Navigator>
      <Stack.Screen name="Login" component={LoginScreen} />
      <Stack.Screen name="Home" component={HomeScreen} />
    </Stack.Navigator>
  );
}

export default function App() {
  return (
    <GestureHandlerRootView style={{ flex: 1 }}>
      <NavigationContainer>
        <MyStack />
      </NavigationContainer>
    </GestureHandlerRootView>
  );
}

const styles = StyleSheet.create({
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  body: {
    backgroundColor: Colors.white,
  },
});
```
</details>

**Comment by t0ma5h** [comment]
Same issues. Still not resolved in 4.16.0. Using with expo sdk 54.
